app/Decsys/Services/ComponentService.cs

```diff
@@ -38,9 +38,7 @@ public ComponentService(LiteDbFactory db, IMapper mapper, ImageService images)
         /// <exception cref="KeyNotFoundException">If the Survey or Page cannot be found.</exception>
         public Models.Component Create(int id, Guid pageId, string type)
         {
-            var surveys = _db.GetCollection<Survey>(Collections.Surveys);
-            var survey = surveys.FindById(id)
-                ?? throw new KeyNotFoundException();
+            var survey = _components.Get(id);
```

Comment:
this is a fine approach (assuming it's working) but if this is the way we're going, then you don't need `IComponentRepository` at all anymore - just get the survey from the `ISurveyRepository` instead, as that's what it's for.

Reply:
Only issue with just using the ISurveyRepo to get the survey is that they return models rather than entities... so I would have to map them to entities. I wasn't sure which way made more sense.

Reply:
But the service can't deal with entities, because it doesn't know what database provider (and therefore what entities) is being used. It has to use models.